command.go

```diff
@@ -370,6 +371,9 @@ func (c *Command) Find(arrs []string) (*Command, []string, error) {
 				// only accept a single prefix match - multiple matches would be ambiguous
 				if len(matches) == 1 {
 					return innerfind(matches[0], argsMinusX(args, argsWOflags[0]))
+				} else if len(matches) == 0 && len(args) > 0 && args[0] == "help" {
+					// special case help command
+					return innerfind(c, argsMinusX(append(args, "--help"), argsWOflags[0]))
```

Comment:
I changed this PR to revert that piece of code (on a separate commit so we don't loose it), since it may result in unexpected behavior not valid for every scenario.

The "help" string could be a valid argument that does not mean help. Using `git add` as an example:

`git add help`

Means I want to add the file called "help", which is perfectly plausible. I *do not* want the equivalent of `git add --help` here.